### DISCUSSION TITLE
Migration from secretstore not working (as expected) - netbox 3.4.7

### DISCUSSION THREAD

**Discussion by greinick** [Onemind-Services-LLC/netbox-secrets#52] [Q&A] [answered]
Hi,

First of all thanks for developing and sharing such a nice add on for netbox!

Hopefully you have an Idea what might be wrong with my setup?

Today I wanted to do an update of netbox , so first I wanted to migrate from secretstore to secrets and followed your documentation:

---
vim /opt/netbox-3.4.7/local_requirements.txt
add netbox-secrets

run ./venv/bin/pip install -r local_requirements.txt

**Output:**
Installing collected packages: netbox-secrets
Successfully installed netbox-secrets-1.7.6

---
vim /opt/netbox/netbox/netbox/configuration.py 
add PLUGINS = ['netbox_secretstore','netbox_secrets',]

run pip install git+https://github.com/Onemind-Services-LLC/netbox-secretstore@migration/nb34

**Output:**
Successfully installed importlib-1.0.4 netbox-secretstore-1.4.4 pycryptodome-3.17

---
run ./venv/bin/python netbox/manage.py migrate

**Output:**
SystemCheckError: System check identified some issues:

ERRORS:
netbox_secrets.Secret.tags: (fields.E304) Reverse accessor 'Tag.secret_set' for 'netbox_secrets.Secret.tags' clashes with reverse accessor for 'netbox_secretstore.Secret.tags'.
	HINT: Add or change a related_name argument to the definition for 'netbox_secrets.Secret.tags' or 'netbox_secretstore.Secret.tags'.
netbox_secrets.SecretRole.tags: (fields.E304) Reverse accessor 'Tag.secretrole_set' for 'netbox_secrets.SecretRole.tags' clashes with reverse accessor for 'netbox_secretstore.SecretRole.tags'.
	HINT: Add or change a related_name argument to the definition for 'netbox_secrets.SecretRole.tags' or 'netbox_secretstore.SecretRole.tags'.
netbox_secrets.UserKey.user: (fields.E304) Reverse accessor 'User.user_key' for 'netbox_secrets.UserKey.user' clashes with reverse accessor for 'netbox_secretstore.UserKey.user'.
	HINT: Add or change a related_name argument to the definition for 'netbox_secrets.UserKey.user' or 'netbox_secretstore.UserKey.user'.
netbox_secrets.UserKey.user: (fields.E305) Reverse query name for 'netbox_secrets.UserKey.user' clashes with reverse query name for 'netbox_secretstore.UserKey.user'.
	HINT: Add or change a related_name argument to the definition for 'netbox_secrets.UserKey.user' or 'netbox_secretstore.UserKey.user'.
netbox_secretstore.Secret.tags: (fields.E304) Reverse accessor 'Tag.secret_set' for 'netbox_secretstore.Secret.tags' clashes with reverse accessor for 'netbox_secrets.Secret.tags'.
	HINT: Add or change a related_name argument to the definition for 'netbox_secretstore.Secret.tags' or 'netbox_secrets.Secret.tags'.
netbox_secretstore.SecretRole.tags: (fields.E304) Reverse accessor 'Tag.secretrole_set' for 'netbox_secretstore.SecretRole.tags' clashes with reverse accessor for 'netbox_secrets.SecretRole.tags'.
	HINT: Add or change a related_name argument to the definition for 'netbox_secretstore.SecretRole.tags' or 'netbox_secrets.SecretRole.tags'.
netbox_secretstore.UserKey.user: (fields.E304) Reverse accessor 'User.user_key' for 'netbox_secretstore.UserKey.user' clashes with reverse accessor for 'netbox_secrets.UserKey.user'.
	HINT: Add or change a related_name argument to the definition for 'netbox_secretstore.UserKey.user' or 'netbox_secrets.UserKey.user'.
netbox_secretstore.UserKey.user: (fields.E305) Reverse query name for 'netbox_secretstore.UserKey.user' clashes with reverse query name for 'netbox_secrets.UserKey.user'.
	HINT: Add or change a related_name argument to the definition for 'netbox_secretstore.UserKey.user' or 'netbox_secrets.UserKey.user'.

**Comment by abhi1693** [comment]
All installations must be within the virtual env. Your error only means the secretstore migration branch is not found in your virtual env

**Reply by greinick** [reply]
Thanks for that fast reply ... :) ! as I'm not so familiar with virtual environment and python ... how do I "switch" or use the venv? Thx.

**Reply by abhi1693** [reply, marked as answer]
Add `git+https://github.com/Onemind-Services-LLC/netbox-secretstore@migration/nb34` to your `local_requirements.txt` and run the install. Since, everyone has a different setup, so I cannot provide all the different ways one can acheive the migration tasks. Depending on your setup, you just have to use the guide as a reference and work your way into it

**Reply by greinick** [reply]
Thanks for the hint. Worked now :) !

**Reply by lill-peter** [reply]
Hello, I'am on the same version of Netbox v3.4.7 and when i try to run 
./venv/bin/python netbox/manage.py migrate
I get this at the end: 
django.core.exceptions.ImproperlyConfigured: Plugin netbox_secrets requires NetBox minimum version 3.5.0.
Do you have any ideas for a solution?

**Reply by abhi1693** [reply]
You need to install v1.7.6 as the v1.8.x is not compatible with older netbox version. Please see the README for more details https://github.com/Onemind-Services-LLC/netbox-secrets#compatibility

**Reply by lill-peter** [reply]
Sorry for this. But how do I install version v1.7.6? I'm, a rookie on github.  
Cheers

**Reply by lill-peter** [reply]
I finaly got the migration right but then I get this when i try to add a new secret

<class 'django.db.utils.IntegrityError'>
duplicate key value violates unique constraint "netbox_secrets_secret_pkey"
DETAIL:  Key (id)=(10) already exists.

Python version: 3.8.10
NetBox version: 3.5.2
Netbox_secrets: 1.8.2

Can I put the start ID somewhere or is there any other solution?

**Reply by abhi1693** [reply]
Migration for v3.5 is not supported. It should only be done via NetBox v3.4 as there is no secretstore plugin on v3.5. Also, it looks like you have had data in the tables for secrets before running the migration. This was pointed out the FAQ section that if migration is desired this should be empty tables for secret.

**Reply by Dominic-Ouellette** [reply]
> Sorry for this. But how do I install version v1.7.6? I'm, a rookie on github. Cheers

Put netbox-secrets==1.7.6 into local_requirements.txt
